### PR TITLE
Add doc page for capture checking.

### PR DESCRIPTION
**Pull request by odersky**
Add doc page that explains what capture checking is, emphasizing that
it is still a research project in a separate branch.

### CODE REVIEW

File: docs/docs/reference/experimental/cc.md [hunk 1]
```diff
@@ -0,0 +1,716 @@
...
+   of each parent class of `C`.
+ - The type of `this` must observe all constraints where `this` is used.
+
+For instance, in
+```scala
+@capability class Cap
+def test(c: Cap) =
+  class A:
+    val x: A = this
+    def f = println(c)  // error
```
Comment (smarter) [inline docs/docs/reference/experimental/cc.md:337]: I'm not completely clear on this example, if `f` was defined before `x`, would that change the inferred type for `this` and lead to an error when typing `x`?

File: docs/docs/reference/experimental/cc.md [hunk 2]
```diff
@@ -0,0 +1,716 @@
...
+```
+Here is a formulation of the class for lazy cons nodes:
+```scala
+import scala.compiletime.uninitialized
+
+final class LzyCons[+A](hd: A, tl: () => {*} LzyList[A]) extends LzyList[A]:
+  private var forced = false
+  private var cache: {this} LzyList[A] = uninitialized
+  private def force =
+    if !forced then { cache = tl(); forced = true }
```
Comment (smarter) [inline docs/docs/reference/experimental/cc.md:550]: Does this mean that `{*} LzyList[A] <: {this} LzyList[A]` ? How does this follow exactly?

File: docs/docs/reference/experimental/cc.md (hunk 1 above)
Comment (odersky) [reply docs/docs/reference/experimental/cc.md:337]: Yes. You get:
```
16 |    val x: A = this
   |               ^^^^
   |               Found:    (A.this : {c} A)
   |               Required: A
```

File: docs/docs/reference/experimental/cc.md (hunk 2 above)
Comment (odersky) [reply docs/docs/reference/experimental/cc.md:550]: No, it's more subtle. See https://github.com/lampepfl/dotty/discussions/14387

I still need to work that in the discussion here somehow.

**Pull request merged**